mm.c

```diff
@@ -366,6 +417,111 @@ static void test_MAIN_FREE_LIST_INIT()
     printf("Test passed.\n\n");
 }
 
+static void test_PREV_NEXT_IN_HEAP() {
+	printf("Testing PREV_NEXT_IN_HEAP.\n");
+
+	size_t prev_size = 4;
+	size_t curr_size = 10;
+	size_t total_needed_test_size = prev_size + curr_size + HDR_FTR_SIZE*2;
+
+	char **ptr = malloc(WORD_SIZE * total_needed_test_size);
+	char **p_prev = ptr;
+	char **p_curr = ptr + HDR_FTR_SIZE + prev_size;
+
+	PUT_WORD(p_prev, PACK(prev_size, FREE));
+	PUT_WORD(FTRP(p_prev), PACK(prev_size, FREE));
+
+	PUT_WORD(p_curr, PACK(curr_size, FREE));
+	PUT_WORD(FTRP(p_curr), PACK(curr_size, FREE));
+
+	char **tmp = ((char **)(p_curr) - GET_SIZE((char **)(p_curr) - 1) - HDR_FTR_SIZE);
+
+	assert(p_prev == PREV_BLOCK_IN_HEAP(p_curr));
+	assert(p_curr == NEXT_BLOCK_IN_HEAP(p_prev));
+
+	printf("Test passed.\n\n");
+}
+
+static void test_coalesce() {
```

Comment:
In the interest of bug free code, we should also test that the blocks that are being coalesced have been removed from their respective free blocks. 

This means, we need functions to insert free block, remove free block, only then we can safely test coalesce. I suggest that we do those first, before finishing this PR.

I can take the insert function, do you want to take the remove function. WDYT ?